### PR TITLE
S04 cat display

### PR DESCRIPTION
**Pull request by Maachino**
Create videocontext , props limited 
Fetch data on videocontext
Editing Home 
make CategoryPage active

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -9,21 +9,27 @@ import Login from "./pages/Login";
...
-          </Routes>
+          <VideoProvider>
+            <Routes>
+              <Route path="/" element={<Home />} />
+              <Route
+                path="/category/:categoryName"
+                element={<CategoryPage />}
+              />
+              <Route path="/search/" element={<Search />} />
+              <Route path="/connexion" element={<Login />} />
```
Comment (rocambille) [inline frontend/src/App.jsx]: in english

File: frontend/src/contexts/VideoContext.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  useEffect(() => {
+    fetch(
+      `${import.meta.env.VITE_BACKEND_URL ?? "http://localhost:5000"}/videos`
+    )
+      .then((response) => response.json())
+      .then((data) => setVideos(data));
+  }, []);
+
+  const value = useMemo(() => [videos, setVideos], [videos]);
```
Comment (rocambille) [inline frontend/src/contexts/VideoContext.jsx]: prefer an object here:

    const value = useMemo(() => {videos, setVideos}, [videos]);

File: frontend/src/pages/CategoryPage.jsx [hunk 3]
```diff
@@ -1,13 +1,45 @@
...
+      categoryVideos = shuffledVideos.slice(0, 3);
+    }
+    for (let i = 0; i < 3; i += 1) {
+      categoryVideos.push(...categoryVideos);
+    }
+    return categoryVideos.slice(0, 15);
+  };
+
+  const videoElements = getVideosByCategory(categoryId).map((video, index) => (
+    <div key={`${video.id}+${index * 4}`}>
```
Comment (rocambille) [inline frontend/src/pages/CategoryPage.jsx]: video.id is enough

**Review by rocambille — CHANGES_REQUESTED** [review]

File: frontend/src/App.jsx (hunk 1 above)
Comment (Maachino) [reply frontend/src/App.jsx]: Done

File: frontend/src/contexts/VideoContext.jsx (hunk 2 above)
Comment (Maachino) [reply frontend/src/contexts/VideoContext.jsx]: Done

File: frontend/src/pages/CategoryPage.jsx (hunk 3 above)
Comment (Maachino) [reply frontend/src/pages/CategoryPage.jsx]: Done

**Pull request merged**